pkg/i2gw/providers/istio/converter.go

```diff
@@ -235,3 +249,415 @@ func (c *converter) convertGateway(gw *istioclientv1beta1.Gateway, fieldPath *fi
+				backendRefs = append(backendRefs, gatewayv1.HTTPBackendRef{
+					BackendRef: gatewayv1.BackendRef{
+						BackendObjectReference: *backendObjRef,
+						Weight:                 &routeDestination.Weight,
+					},
+				})
+			}
+		}
+
+		if routeRedirect := httpRoute.GetRedirect(); routeRedirect != nil {
+			redirectFieldPath := httpRouteFieldPath.Child("HTTPRedirect")
+
+			if routeRedirect.GetAuthority() != "" {
+				klog.Infof("ignoring field: %v", redirectFieldPath.Child("Authority"))
+			}
+			if _, ok := routeRedirect.GetRedirectPort().(*istiov1beta1.HTTPRedirect_DerivePort); ok {
+				klog.Infof("ignoring field: %v", redirectFieldPath.Child("DerivePort"))
+			}
+
+			redirectCode := 301
+			if routeRedirect.GetRedirectCode() > 0 {
+				redirectCode = int(routeRedirect.GetRedirectCode())
+			}
+
+			var redirectPath *gatewayv1.HTTPPathModifier
+
+			if routeRedirectURI := routeRedirect.GetUri(); routeRedirectURI != "" {
+				redirectPath = &gatewayv1.HTTPPathModifier{
+					Type:            gatewayv1.FullPathHTTPPathModifier,
+					ReplaceFullPath: &routeRedirectURI,
+				}
+			}
+
+			redirectFilter := gatewayv1.HTTPRequestRedirectFilter{
+				StatusCode: &redirectCode,
+				Path:       redirectPath,
+			}
+
+			if routeRedirectScheme := routeRedirect.GetScheme(); routeRedirectScheme != "" {
+				redirectFilter.Scheme = &routeRedirectScheme
+			}
+
+			if routeRedirect.GetPort() > 0 {
+				redirectPort := gatewayv1.PortNumber(routeRedirect.GetPort())
+				redirectFilter.Port = &redirectPort
+			}
+
+			gwHTTPRouteFilters = append(gwHTTPRouteFilters, gatewayv1.HTTPRouteFilter{
+				Type:            gatewayv1.HTTPRouteFilterRequestRedirect,
+				RequestRedirect: &redirectFilter,
+			})
+		}
+
+		if httpRoute.GetDirectResponse() != nil {
+			klog.Infof("ignoring field: %v", httpRouteFieldPath.Child("DirectResponse"))
+		}
+		if httpRoute.GetDelegate() != nil {
+			klog.Infof("ignoring field: %v", httpRouteFieldPath.Child("Delegate"))
+		}
+		if httpRoute.GetRetries() != nil {
+			klog.Infof("ignoring field: %v", httpRouteFieldPath.Child("Retries"))
+		}
+		if httpRoute.GetFault() != nil {
+			klog.Infof("ignoring field: %v", httpRouteFieldPath.Child("Fault"))
+		}
+		if httpRoute.GetCorsPolicy() != nil {
+			klog.Infof("ignoring field: %v", httpRouteFieldPath.Child("CorsPolicy"))
+		}
+
+		if rewrite := httpRoute.GetRewrite(); rewrite != nil {
+			rewriteFieldPath := httpRouteFieldPath.Child("HTTPRewrite")
+
+			if rewrite.GetAuthority() != "" {
+				klog.Infof("ignoring field: %v", rewriteFieldPath.Child("Authority"))
+			}
+			if rewrite.GetUriRegexRewrite() != nil {
+				klog.Infof("ignoring field: %v", rewriteFieldPath.Child("UriRegexRewrite"))
+			}
+
+			gwHTTPRouteFilters = append(gwHTTPRouteFilters, gatewayv1.HTTPRouteFilter{
+				Type: gatewayv1.HTTPRouteFilterURLRewrite,
+				URLRewrite: &gatewayv1.HTTPURLRewriteFilter{
+					Path: &gatewayv1.HTTPPathModifier{
+						Type:               gatewayv1.PrefixMatchHTTPPathModifier,
```

Comment:
istio also supports full `gatewayv1.FullPathHTTPPathModifier`.  (https://github.com/istio/istio/pull/42605)

The case where it would be relevant is based on how the request was matched (https://istio.io/latest/docs/reference/config/networking/virtual-service/#HTTPRewrite the match field.)

So I think exact match and regex match would need  gatewayv1.FullPathHTTPPathModifier and not PrefixMatchHTTPPathModifier

Reply:
Add `rewrite` handling logic: https://github.com/kubernetes-sigs/ingress2gateway/pull/111/commits/e39d712fe2b451707f54e739c77a2b64f0983dc8